### ISSUE TITLE
URL doesn't restore state when the field is a SelectInputList with multiple=T where list has only one member

### ISSUE THREAD

**Issue by dividor** [aoles/shinyURL#5]
Hi Aoles,

Thanks for this great package, exactly what I was looking for!

I do seem to have an issue however. If my UI has selectizeInput fields which allow multiple sections (multiple=T), but only one value in the field, then the URL generated by ShinyURL doesn't fully restore the session. I think it's because that for list fields, they require the array reference, even if only one member.

So for example, here is my field definition:

selectizeInput("filter_choices", "Active Filters", filterChoiceList,
                         options=list(placeholder="Choose fields to filter on"),
                         multiple = T)

Here is the URL generated if I only have one value in "filter_choices" ...

http://127.0.0.1:6777/?yval=Assignee_Group&Assignee=&metric=average_age&dates=16437,16853&date_type=opened_between&filter_choices=business_unit&business_unit=CAT&xval=ClosedMonth&metric_cutoff=0,178

This doesn't work. However, if I manually edit it to include "[1]" for the list fields with only one value ...

http://127.0.0.1:6777/?yval=Assignee_Group&Assignee=&metric=average_age&dates=16437,16853&date_type=opened_between&filter_choices[1]=business_unit&business_unit[1]=CAT&xval=ClosedMonth&metric_cutoff=0,178

This works perfectly and restores the Shiny session.

Note, that if there are multiple choices in these list fields everything works perfectly, the issue is limited to when there is only one value in a multiple selection list field. 

In summary, I think perhaps the generated URL needs the "[1]" when only one value is specified for multiple=T selectInput fields.

Thanks!

**Comment by aoles** [comment]
Thank you @dividor for your detailed report!

I checked this using the example distributed with the package (see the "Multiple select" box)

```
shinyAppDir( system.file('examples', 'widgets', package='shinyURL') )
```

but couldn't reproduce the described behaviour. Could it be that you are using an old version of shinyURL? This might have been already fixed in https://github.com/aoles/shinyURL/commit/77ca91ab2d05c980104709debfc6336eae2ca4c8.

**Comment by dividor** [comment]
Hi @aoles,

I'm using 0.0.31.

**Comment by aoles** [comment]
@dividor Can you please provide a minimal reproducible example illustrating the issue? Thanks!
